designdocs/provider_registration.md

```diff
@@ -0,0 +1,175 @@
+included in the announcement must already exist.
+
+This extrinsic is responsible for storing the registered provider in the
+`ProviderRegistry`.
+
+* **Parameters**
+  * `origin`: `Origin`  required for all extrinsics, the caller/sender.
+  * `registration_params`: `ProviderRegistrationParams`, the parameters to use in the registration announcement.
+* **Event**:  `Event::<T>::ProviderRegistered(provider_msa_id, provider_metadata)`
+* **Restrictions**:
+  * `origin`'s `msa_id` must have capacity to post the transaction (including fee) during the current epoch
+
+#### unregister_provider(origin, unregistration_params)
+Creates and posts a `ProviderUnregistrationAnnouncement` on chain. The `MsaId`
+included in the announcement must already exist.
+
+This extrinsic is responsible for deleting the registered provider's `MsaId` from the
+`ProviderRegistry`.
+
+* **Parameters**
+  * `origin`: `Origin`  required for all extrinsics, the caller/sender.
+  * `unregistration_params`: `ProviderUnregistrationParams`, the parameters to use in the unregistration announcement.
+* **Event**:  `Event::<T>::ProviderUnregistered(provider_msa_id, provider_name)`
+* **Restrictions**:
+  * `origin`'s `msa_id` must have capacity to post the transaction during the current epoch
+
+### Custom RPCs
+#### get_provider_announcement(provider_msa_id)
+Retrieves a single provider announcement. The `provider_msa_id` can belong to a
+registered or unregistered provider.
+
+* **Parameters**
+  * `provider_msa_id`: `MsaId` the `MsaId` of the provider in question.
+
+* **Returns**
+  * `None()` if no messages meet the criteria.
+  * `Some(ProviderRegistrationAnnouncement)`
+
+#### get_provider_announcements(options)
+Retrieves paged registration announcements that have not been garbage-collected
+which meet `options` criteria. The `provider_msa_id`s can belong to
+registered or unregistered providers.
+
+* **Parameters**
+  * `options`: `ProviderRegistrationAnnouncementOptions` containing regitration announcement criteria
+
+* **Returns**
+  * `None()` if no messages meet the criteria.
+  * `Some(Vec<ProviderRegistrationAnnouncement>)`, in descending block-transaction order
+
+## Benefits and Risks
+### Provider Registry and Info
+This structure allows for easy provider lookups, in the event actors on the
+chain need to query provider information. As of the time of writing this
+document, the only provider information required is a name (non-unique). In the
+future, we may include other information like domain, logo, business address, etc.
+
+### Provider Fees
+This document assumes that registration fees will be a fixed amount. It may be
+prudent, in the future, to determine whether or not registrations are fixed or
+vary based on the amount of verifable information given on a registration
+attempt. The latter may allow us to place more trust into providers that do not
+have traditional business attributes.
```

Comment:
@wilwade thoughts on this language? Especially section about using variable registration fees to gauge "risk" of trusting a provider.